### DISCUSSION TITLE
Block prod dependency release until dev dependency is merged in monorepo

### DISCUSSION THREAD

**Discussion by ChipWolf** [renovatebot/renovate#14112] [Request Help]
We have a monorepo with a folder in the root for the dev environment and another for the prod environment.

We have groups set up in Renovate so each environment has different release PRs.
Each environment also has different stability days settings.

We'd very much like to prevent the prod dependency release from being merged before the dev one.
Is this possible without any McGyvering?

**Comment by rarkins** [comment]
Unfortunately nothing I can think of. You'd need the allowedVersions of the prod dependency to somehow be <= the current version of the corresponding dev dependency?

**Reply by ChipWolf** [reply]
That gives me an idea, I could Renovate the Renovate config during the dev release _(very meta)_ on the prod package rule `allowedVersions` field.

Unless you have another idea @rarkins?

**Reply by ChipWolf** [reply]
```json5
 "regexManagers": [
    {
      "fileMatch": ["./.*\\.yaml$", "./renovate\.json$"],
      "matchStrings": [
        "OUR_PACKAGE: \"?(?<currentValue>.*).zip\"?\n",
        "allowedVersions\": \"\<=(?<currentValue>.*)\""
      ],
      "depNameTemplate": "our-org/our-package",
      "datasourceTemplate": "github-releases"
    }
  ],
  "packageRules": [
    {
      "groupName": "dev",
      "matchPackagePatterns": ["our-package"],
      "matchPaths": ["dev/**", "renovate.json"],
      "stabilityDays": 0
    },
    {
      "groupName": "prod",
      "matchPackagePatterns": ["our-package"],
      "matchPaths": ["prod/**"],
      "stabilityDays": 3,
      "allowedVersions": "<=1.0.0"
    }
  ]
```

**Reply by rarkins** [reply]
I have no ideas.. but also don't understand your proposal. If the `allowedVersions` in prod is hard-coded then how would that work?

**Reply by ChipWolf** [reply]
> I have no ideas.. but also don't understand your proposal. If the `allowedVersions` in prod is hard-coded then how would that work?

The dev release would bump the prod allowed version, see the second match string.

**Reply by dudicoco** [reply]
@ChipWolf you could try using https://probot.github.io/apps/dep/ to prevent prod from being merged before dev.

**Reply by ChipWolf** [reply]
> @ChipWolf you could try using https://probot.github.io/apps/dep/ to prevent prod from being merged before dev.

Hey, thanks for this @dudicoco. We're now using dep, but unfortunately, it doesn't help with this use-case as it expects the 'Depends on...' string to be in the PR description, which Renovate reverts.

Ideally, I'd like something that is completely automated. Our use-case has expanded now and the allowedVersions solution above doesn't scale well.

**Reply by dudicoco** [reply]
> Hey, thanks for this @dudicoco. We're now using dep, but unfortunately, it doesn't help with this use-case as it expects the 'Depends on...' string to be in the PR description, which Renovate reverts.

@ChipWolf what do you mean? How can Renovate revert a comment?

**Reply by ChipWolf** [reply]
> > Hey, thanks for this @dudicoco. We're now using dep, but unfortunately, it doesn't help with this use-case as it expects the 'Depends on...' string to be in the PR description, which Renovate reverts.
> 
> 
> 
> @ChipWolf what do you mean? How can Renovate revert a comment?

Not comments, the PR description (the PR body)

**Reply by dudicoco** [reply]
@ChipWolf in which cases does renovate reverts the PR description? When it rebases/updates the PR?

**Reply by rarkins** [reply]
Renovate updates mismatched PR body (according to what it thinks it should be) on every run.

**Reply by dudicoco** [reply]
@rarkins is there a configuration option to make renovate ignore changes to the PR body?

**Reply by rarkins** [reply]
No, that would be impractical and cause some undesirable side effects such as PR body nor matching branch contents

**Reply by ChipWolf** [reply]
Can we convert this into a feature request @rarkins? I imagine something like this:
```
  "packageRules": [
    {
      "group": "dev",
      "matchPaths": ["dev/**"],
    },
    {
      "group": "prod",
      "dependsOnGroup": "dev,
      "matchPaths": ["prod/**"],
    },
  ]
```

**Reply by rarkins** [reply]
You can, but right now I don't like it and think it would add undesirable complexity for a very narrow use case. But maybe I don't understand it fully, you can feel free to convince me. Please be sure to back up to the full requirements and motivations when describing and not just the proposed solution, so that we can work out if there's any elegant way to do it.
